common/djangoapps/util/milestones_helpers.py

```diff
@@ -351,12 +341,31 @@ def get_course_content_milestones(course_id, content_id, relationship, user_id=N
     """
     if not settings.FEATURES.get('MILESTONES_APP', False):
         return []
-    from milestones import api as milestones_api
+
+    if user_id is None:
+        return milestones_api.get_course_content_milestones(course_id, content_id, relationship)
+
+    request_cache_dict = request_cache.get_cache(REQUEST_CACHE_NAME)
+    if user_id not in request_cache_dict:
+        request_cache_dict[user_id] = milestones_api.get_course_content_milestones(
+            course_key=course_id,
+            user={"id": user_id}
+        )
+    milestones_for_content = []
+    if relationship == "requires":
+        for milestone in request_cache_dict[user_id]:
+            if milestone["content_id"] == content_id and milestone["requirements"]:
+                milestones_for_content.append(milestone)
```

Comment:
Stylistically, the `filter` function is frowned on by much of the python community, in favor of list/generator comprehensions (I personally don't mind it, but just so you're aware).